src/target/riscv/riscv.c

```diff
@@ -5698,19 +5779,21 @@ static int check_if_trigger_exists(struct target *target, unsigned int index)
  * to determine trigger types supported by a trigger.
  * It is assumed that the trigger is already selected via writing `tselect`.
  */
-static int get_trigger_types(struct target *target, unsigned int *trigger_tinfo,
-		riscv_reg_t tdata1)
+static int get_trigger_types_and_version(struct target *target, unsigned int *trigger_tinfo,
+		int64_t *trigger_tinfo_version, riscv_reg_t tdata1)
 {
 	assert(trigger_tinfo);
 	riscv_reg_t tinfo;
 	if (riscv_reg_get(target, &tinfo, GDB_REGNO_TINFO) == ERROR_OK) {
+		*trigger_tinfo_version = get_field(tinfo, CSR_TINFO_VERSION);
 		/* tinfo.INFO == 1: trigger doesn’t exist
 		 * tinfo == 0 or tinfo.INFO != 1 and tinfo LSB is set: invalid tinfo */
 		if (tinfo == 0 || tinfo & 0x1)
 			return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
 		*trigger_tinfo = tinfo;
 		return ERROR_OK;
 	}
+	*trigger_tinfo_version = ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
```

Comment:
Please, don't reuse OpenOCD return codes to denote "trigger not available". Instead, make a dedicated constant for it, e.g. `#define RISCV_TINFO_VERSION_UNKNOWN (-1)`.

Reply:
addressed